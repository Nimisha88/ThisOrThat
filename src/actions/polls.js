export const RECEIVE_POLLS = "RECEIVE_POLLS";
export const REGISTER_VOTE = "REGISTER_VOTE";

export const receivePolls = (polls) => {
    return {
        type: RECEIVE_POLLS,
        polls
    }
}

export const registerVoteInPolls = (vote) => {
    return {
        type: REGISTER_VOTE,
        vote
    }
}

// Action Creators