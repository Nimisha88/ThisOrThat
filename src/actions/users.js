export const RECEIVE_USERS = "RECEIVE_USERS";
export const REGISTER_VOTE = "REGISTER_VOTE";

export const receiveUsers = (users) => {
    return {
        type: RECEIVE_USERS,
        users
    }
}

export const registerVoteInUsers = (vote) => {
    return {
        type: REGISTER_VOTE,
        vote
    }
}

// Action Creators