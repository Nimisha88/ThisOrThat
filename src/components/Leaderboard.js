import { connect } from "react-redux";

const Leaderboard = ({ dispatch }) => {
    return (
        <div>
            Leaderboard
        </div>
    );
}

const mapStatesToProps = ({ polls, users, authedUser }) => {
    return {

    }
}

export default connect(mapStatesToProps)(Leaderboard);