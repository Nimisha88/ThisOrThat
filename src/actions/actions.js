import { getInitialData, savePollAnswer, savePoll } from "../utils/api"
import { receivePolls, registerVoteInPolls, addPollInPolls } from "./polls"
import { receiveUsers, registerVoteInUsers, addPollInUsers } from "./users";
import { setAuthedUser } from "./authedUser";
import { showLoading, hideLoading } from "react-redux-loading-bar";


export const initializeAppData = () => {
    return (
        (dispatch) => {
            dispatch(showLoading());
           return getInitialData()
            .then(({ polls, users }) => {
                dispatch(receivePolls(polls));
                dispatch(receiveUsers(users));
                dispatch(setAuthedUser(null));
                dispatch(hideLoading());
            });
        }
    )
}

export const handleVote = (qid, answer) => {
    return (
        (dispatch, getState) => {
            dispatch(showLoading());
            const { authedUser } = getState();
            const vote = {
                authedUser,
                qid,
                answer
            }
            return savePollAnswer(vote)
            .then((isResolved) => {
                if(isResolved) {
                    dispatch(registerVoteInPolls(vote));
                    dispatch(registerVoteInUsers(vote));
                    dispatch(hideLoading());
                }
            }).catch((error) => {
                alert("Oops! There was an error to save your vote, please try again later!");
                console.log(error);
            }); 
        }
    )
}

export const handleCreatePoll = (poll) => {
    return (
        (dispatch) => {
            dispatch(showLoading());
            return savePoll(poll)
            .then((formattedPoll) => {
                alert("Awesome! Your poll was saved!");
                dispatch(addPollInPolls(formattedPoll));
                dispatch(addPollInUsers(formattedPoll));
                dispatch(hideLoading());
            }).catch((error) => {
                alert("Oops! There was an error to save your poll, please try again later!");
                console.log(error);
            });
        }
    )
}