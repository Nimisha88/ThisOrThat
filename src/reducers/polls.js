import { RECEIVE_POLLS, REGISTER_VOTE } from "../actions/polls"

const polls = (state = {}, action) => {
    switch(action.type) {
        case RECEIVE_POLLS:
            return {
                ...state,
                ...action.polls
            }
        case REGISTER_VOTE:
            const { authedUser, qid, answer } = action.vote;
            return {
                ...state,
                [qid]: {
                    ...state[qid],
                    optionOne: {
                        ...state[qid].optionOne,
                        votes: answer==="optionOne"? state[qid].optionOne.votes.concat(authedUser)
                        : state[qid].optionOne.votes
                    },
                    optionTwo: {
                        ...state[qid].optionTwo,
                        votes: answer==="optionTwo"? state[qid].optionTwo.votes.concat(authedUser)
                        : state[qid].optionTwo.votes
                    }
                }
            }
        default:
            return state;
    }
}

export default polls;