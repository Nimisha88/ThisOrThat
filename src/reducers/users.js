import { RECEIVE_USERS, REGISTER_VOTE } from "../actions/users"

const users = (state = {}, action) => {
    switch(action.type) {
        case RECEIVE_USERS:
            return {
                ...state,
                ...action.users
            }
        case REGISTER_VOTE:
            const { authedUser, qid, answer } = action.vote;
            return {
                ...state,
                [authedUser]: {
                    ...state[authedUser],
                    answers: {
                        ...state[authedUser].answers,
                        [qid]: answer
                    }
                }
            }
        default:
            return state;
    }
}

export default users;