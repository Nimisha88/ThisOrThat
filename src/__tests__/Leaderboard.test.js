import { render } from "@testing-library/react";
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducers from "../reducers/reducers"
import middlewares from "../middlewares/middlewares"
import { MemoryRouter as Router } from 'react-router-dom';
import { initializeAppData } from "../actions/actions";
import { setAuthedUser } from "../actions/authedUser";
import Leaderboard from "../components/Leaderboard";

const store = createStore(reducers, middlewares);

test("renders Leaderboard", async () => {
    await store.dispatch(initializeAppData());
    await store.dispatch(setAuthedUser("sarahedo"));
    const view = render(
        <Provider store={store}>
            <Router>
                <Leaderboard />
            </Router>
        </Provider>
    );
    
    expect(view).toMatchSnapshot();
});