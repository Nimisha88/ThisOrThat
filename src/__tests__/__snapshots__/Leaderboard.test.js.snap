// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="leaderboard"
      >
        <h2
          class="subtitle"
        >
          Leaderboard
        </h2>
        <h3>
          Winner: 
          <span
            class="winner"
          >
            Sarah Edo
          </span>
        </h3>
        <div
          class="lbcards-container"
        >
          <div
            class="lb-card"
          >
            <div
              class="lbcard-avatar"
            >
              <div
                class="avatar sb-avatar"
                style="display: inline-block; vertical-align: middle; width: 70px; height: 70px; border-radius: 50%; font-family: Helvetica, Arial, sans-serif;"
              >
                <div
                  class="avatar sb-avatar__text"
                  style="width: 70px; height: 70px; line-height: initial; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
                  title="Sarah Edo"
                >
                  <div
                    style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                  >
                    <span
                      style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                    >
                      <span />
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="do-nothing"
            />
            <div
              class="lbcard-user"
            >
              <h3>
                Sarah Edo
              </h3>
              <h5>
                sarahedo
              </h5>
            </div>
            <div
              class="do-nothing"
            />
            <div
              class="lbcard-score"
            >
              <h4>
                Asked: 
                2
              </h4>
              <h4>
                Answered: 
                4
              </h4>
              <h4
                class="score"
              >
                Total: 
                6
              </h4>
            </div>
          </div>
          <div
            class="lb-card"
          >
            <div
              class="lbcard-avatar"
            >
              <div
                class="avatar sb-avatar"
                style="display: inline-block; vertical-align: middle; width: 70px; height: 70px; border-radius: 50%; font-family: Helvetica, Arial, sans-serif;"
              >
                <div
                  class="avatar sb-avatar__text"
                  style="width: 70px; height: 70px; line-height: initial; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
                  title="Mike Tsamis"
                >
                  <div
                    style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                  >
                    <span
                      style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                    >
                      <span />
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="do-nothing"
            />
            <div
              class="lbcard-user"
            >
              <h3>
                Mike Tsamis
              </h3>
              <h5>
                mtsamis
              </h5>
            </div>
            <div
              class="do-nothing"
            />
            <div
              class="lbcard-score"
            >
              <h4>
                Asked: 
                2
              </h4>
              <h4>
                Answered: 
                3
              </h4>
              <h4
                class="score"
              >
                Total: 
                5
              </h4>
            </div>
          </div>
          <div
            class="lb-card"
          >
            <div
              class="lbcard-avatar"
            >
              <div
                class="avatar sb-avatar"
                style="display: inline-block; vertical-align: middle; width: 70px; height: 70px; border-radius: 50%; font-family: Helvetica, Arial, sans-serif;"
              >
                <div
                  class="avatar sb-avatar__text"
                  style="width: 70px; height: 70px; line-height: initial; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
                  title="Tyler McGinnis"
                >
                  <div
                    style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                  >
                    <span
                      style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                    >
                      <span />
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="do-nothing"
            />
            <div
              class="lbcard-user"
            >
              <h3>
                Tyler McGinnis
              </h3>
              <h5>
                tylermcginnis
              </h5>
            </div>
            <div
              class="do-nothing"
            />
            <div
              class="lbcard-score"
            >
              <h4>
                Asked: 
                2
              </h4>
              <h4>
                Answered: 
                2
              </h4>
              <h4
                class="score"
              >
                Total: 
                4
              </h4>
            </div>
          </div>
          <div
            class="lb-card"
          >
            <div
              class="lbcard-avatar"
            >
              <div
                class="avatar sb-avatar"
                style="display: inline-block; vertical-align: middle; width: 70px; height: 70px; border-radius: 50%; font-family: Helvetica, Arial, sans-serif;"
              >
                <div
                  class="avatar sb-avatar__text"
                  style="width: 70px; height: 70px; line-height: initial; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
                  title="Zenobia Oshikanlu"
                >
                  <div
                    style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                  >
                    <span
                      style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                    >
                      <span />
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="do-nothing"
            />
            <div
              class="lbcard-user"
            >
              <h3>
                Zenobia Oshikanlu
              </h3>
              <h5>
                zoshikanlu
              </h5>
            </div>
            <div
              class="do-nothing"
            />
            <div
              class="lbcard-score"
            >
              <h4>
                Asked: 
                0
              </h4>
              <h4>
                Answered: 
                1
              </h4>
              <h4
                class="score"
              >
                Total: 
                1
              </h4>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="leaderboard"
    >
      <h2
        class="subtitle"
      >
        Leaderboard
      </h2>
      <h3>
        Winner: 
        <span
          class="winner"
        >
          Sarah Edo
        </span>
      </h3>
      <div
        class="lbcards-container"
      >
        <div
          class="lb-card"
        >
          <div
            class="lbcard-avatar"
          >
            <div
              class="avatar sb-avatar"
              style="display: inline-block; vertical-align: middle; width: 70px; height: 70px; border-radius: 50%; font-family: Helvetica, Arial, sans-serif;"
            >
              <div
                class="avatar sb-avatar__text"
                style="width: 70px; height: 70px; line-height: initial; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
                title="Sarah Edo"
              >
                <div
                  style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                >
                  <span
                    style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                  >
                    <span />
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="do-nothing"
          />
          <div
            class="lbcard-user"
          >
            <h3>
              Sarah Edo
            </h3>
            <h5>
              sarahedo
            </h5>
          </div>
          <div
            class="do-nothing"
          />
          <div
            class="lbcard-score"
          >
            <h4>
              Asked: 
              2
            </h4>
            <h4>
              Answered: 
              4
            </h4>
            <h4
              class="score"
            >
              Total: 
              6
            </h4>
          </div>
        </div>
        <div
          class="lb-card"
        >
          <div
            class="lbcard-avatar"
          >
            <div
              class="avatar sb-avatar"
              style="display: inline-block; vertical-align: middle; width: 70px; height: 70px; border-radius: 50%; font-family: Helvetica, Arial, sans-serif;"
            >
              <div
                class="avatar sb-avatar__text"
                style="width: 70px; height: 70px; line-height: initial; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
                title="Mike Tsamis"
              >
                <div
                  style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                >
                  <span
                    style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                  >
                    <span />
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="do-nothing"
          />
          <div
            class="lbcard-user"
          >
            <h3>
              Mike Tsamis
            </h3>
            <h5>
              mtsamis
            </h5>
          </div>
          <div
            class="do-nothing"
          />
          <div
            class="lbcard-score"
          >
            <h4>
              Asked: 
              2
            </h4>
            <h4>
              Answered: 
              3
            </h4>
            <h4
              class="score"
            >
              Total: 
              5
            </h4>
          </div>
        </div>
        <div
          class="lb-card"
        >
          <div
            class="lbcard-avatar"
          >
            <div
              class="avatar sb-avatar"
              style="display: inline-block; vertical-align: middle; width: 70px; height: 70px; border-radius: 50%; font-family: Helvetica, Arial, sans-serif;"
            >
              <div
                class="avatar sb-avatar__text"
                style="width: 70px; height: 70px; line-height: initial; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
                title="Tyler McGinnis"
              >
                <div
                  style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                >
                  <span
                    style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                  >
                    <span />
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="do-nothing"
          />
          <div
            class="lbcard-user"
          >
            <h3>
              Tyler McGinnis
            </h3>
            <h5>
              tylermcginnis
            </h5>
          </div>
          <div
            class="do-nothing"
          />
          <div
            class="lbcard-score"
          >
            <h4>
              Asked: 
              2
            </h4>
            <h4>
              Answered: 
              2
            </h4>
            <h4
              class="score"
            >
              Total: 
              4
            </h4>
          </div>
        </div>
        <div
          class="lb-card"
        >
          <div
            class="lbcard-avatar"
          >
            <div
              class="avatar sb-avatar"
              style="display: inline-block; vertical-align: middle; width: 70px; height: 70px; border-radius: 50%; font-family: Helvetica, Arial, sans-serif;"
            >
              <div
                class="avatar sb-avatar__text"
                style="width: 70px; height: 70px; line-height: initial; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
                title="Zenobia Oshikanlu"
              >
                <div
                  style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                >
                  <span
                    style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                  >
                    <span />
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="do-nothing"
          />
          <div
            class="lbcard-user"
          >
            <h3>
              Zenobia Oshikanlu
            </h3>
            <h5>
              zoshikanlu
            </h5>
          </div>
          <div
            class="do-nothing"
          />
          <div
            class="lbcard-score"
          >
            <h4>
              Asked: 
              0
            </h4>
            <h4>
              Answered: 
              1
            </h4>
            <h4
              class="score"
            >
              Total: 
              1
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders Leaderboard 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="leaderboard"
      >
        <h2
          class="subtitle"
        >
          Leaderboard
        </h2>
        <h3>
          Winner: 
          <span
            class="winner"
          >
            Sarah Edo
          </span>
        </h3>
        <div
          class="lbcards-container"
        >
          <div
            class="lb-card"
          >
            <div
              class="lbcard-avatar"
            >
              <div
                class="avatar sb-avatar"
                style="display: inline-block; vertical-align: middle; width: 70px; height: 70px; border-radius: 50%; font-family: Helvetica, Arial, sans-serif;"
              >
                <div
                  class="avatar sb-avatar__text"
                  style="width: 70px; height: 70px; line-height: initial; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
                  title="Sarah Edo"
                >
                  <div
                    style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                  >
                    <span
                      style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                    >
                      <span />
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="do-nothing"
            />
            <div
              class="lbcard-user"
            >
              <h3>
                Sarah Edo
              </h3>
              <h5>
                sarahedo
              </h5>
            </div>
            <div
              class="do-nothing"
            />
            <div
              class="lbcard-score"
            >
              <h4>
                Asked: 
                2
              </h4>
              <h4>
                Answered: 
                4
              </h4>
              <h4
                class="score"
              >
                Total: 
                6
              </h4>
            </div>
          </div>
          <div
            class="lb-card"
          >
            <div
              class="lbcard-avatar"
            >
              <div
                class="avatar sb-avatar"
                style="display: inline-block; vertical-align: middle; width: 70px; height: 70px; border-radius: 50%; font-family: Helvetica, Arial, sans-serif;"
              >
                <div
                  class="avatar sb-avatar__text"
                  style="width: 70px; height: 70px; line-height: initial; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
                  title="Mike Tsamis"
                >
                  <div
                    style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                  >
                    <span
                      style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                    >
                      <span />
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="do-nothing"
            />
            <div
              class="lbcard-user"
            >
              <h3>
                Mike Tsamis
              </h3>
              <h5>
                mtsamis
              </h5>
            </div>
            <div
              class="do-nothing"
            />
            <div
              class="lbcard-score"
            >
              <h4>
                Asked: 
                2
              </h4>
              <h4>
                Answered: 
                3
              </h4>
              <h4
                class="score"
              >
                Total: 
                5
              </h4>
            </div>
          </div>
          <div
            class="lb-card"
          >
            <div
              class="lbcard-avatar"
            >
              <div
                class="avatar sb-avatar"
                style="display: inline-block; vertical-align: middle; width: 70px; height: 70px; border-radius: 50%; font-family: Helvetica, Arial, sans-serif;"
              >
                <div
                  class="avatar sb-avatar__text"
                  style="width: 70px; height: 70px; line-height: initial; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
                  title="Tyler McGinnis"
                >
                  <div
                    style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                  >
                    <span
                      style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                    >
                      <span />
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="do-nothing"
            />
            <div
              class="lbcard-user"
            >
              <h3>
                Tyler McGinnis
              </h3>
              <h5>
                tylermcginnis
              </h5>
            </div>
            <div
              class="do-nothing"
            />
            <div
              class="lbcard-score"
            >
              <h4>
                Asked: 
                2
              </h4>
              <h4>
                Answered: 
                2
              </h4>
              <h4
                class="score"
              >
                Total: 
                4
              </h4>
            </div>
          </div>
          <div
            class="lb-card"
          >
            <div
              class="lbcard-avatar"
            >
              <div
                class="avatar sb-avatar"
                style="display: inline-block; vertical-align: middle; width: 70px; height: 70px; border-radius: 50%; font-family: Helvetica, Arial, sans-serif;"
              >
                <div
                  class="avatar sb-avatar__text"
                  style="width: 70px; height: 70px; line-height: initial; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
                  title="Zenobia Oshikanlu"
                >
                  <div
                    style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                  >
                    <span
                      style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                    >
                      <span />
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="do-nothing"
            />
            <div
              class="lbcard-user"
            >
              <h3>
                Zenobia Oshikanlu
              </h3>
              <h5>
                zoshikanlu
              </h5>
            </div>
            <div
              class="do-nothing"
            />
            <div
              class="lbcard-score"
            >
              <h4>
                Asked: 
                0
              </h4>
              <h4>
                Answered: 
                1
              </h4>
              <h4
                class="score"
              >
                Total: 
                1
              </h4>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="leaderboard"
    >
      <h2
        class="subtitle"
      >
        Leaderboard
      </h2>
      <h3>
        Winner: 
        <span
          class="winner"
        >
          Sarah Edo
        </span>
      </h3>
      <div
        class="lbcards-container"
      >
        <div
          class="lb-card"
        >
          <div
            class="lbcard-avatar"
          >
            <div
              class="avatar sb-avatar"
              style="display: inline-block; vertical-align: middle; width: 70px; height: 70px; border-radius: 50%; font-family: Helvetica, Arial, sans-serif;"
            >
              <div
                class="avatar sb-avatar__text"
                style="width: 70px; height: 70px; line-height: initial; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
                title="Sarah Edo"
              >
                <div
                  style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                >
                  <span
                    style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                  >
                    <span />
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="do-nothing"
          />
          <div
            class="lbcard-user"
          >
            <h3>
              Sarah Edo
            </h3>
            <h5>
              sarahedo
            </h5>
          </div>
          <div
            class="do-nothing"
          />
          <div
            class="lbcard-score"
          >
            <h4>
              Asked: 
              2
            </h4>
            <h4>
              Answered: 
              4
            </h4>
            <h4
              class="score"
            >
              Total: 
              6
            </h4>
          </div>
        </div>
        <div
          class="lb-card"
        >
          <div
            class="lbcard-avatar"
          >
            <div
              class="avatar sb-avatar"
              style="display: inline-block; vertical-align: middle; width: 70px; height: 70px; border-radius: 50%; font-family: Helvetica, Arial, sans-serif;"
            >
              <div
                class="avatar sb-avatar__text"
                style="width: 70px; height: 70px; line-height: initial; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
                title="Mike Tsamis"
              >
                <div
                  style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                >
                  <span
                    style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                  >
                    <span />
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="do-nothing"
          />
          <div
            class="lbcard-user"
          >
            <h3>
              Mike Tsamis
            </h3>
            <h5>
              mtsamis
            </h5>
          </div>
          <div
            class="do-nothing"
          />
          <div
            class="lbcard-score"
          >
            <h4>
              Asked: 
              2
            </h4>
            <h4>
              Answered: 
              3
            </h4>
            <h4
              class="score"
            >
              Total: 
              5
            </h4>
          </div>
        </div>
        <div
          class="lb-card"
        >
          <div
            class="lbcard-avatar"
          >
            <div
              class="avatar sb-avatar"
              style="display: inline-block; vertical-align: middle; width: 70px; height: 70px; border-radius: 50%; font-family: Helvetica, Arial, sans-serif;"
            >
              <div
                class="avatar sb-avatar__text"
                style="width: 70px; height: 70px; line-height: initial; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
                title="Tyler McGinnis"
              >
                <div
                  style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                >
                  <span
                    style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                  >
                    <span />
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="do-nothing"
          />
          <div
            class="lbcard-user"
          >
            <h3>
              Tyler McGinnis
            </h3>
            <h5>
              tylermcginnis
            </h5>
          </div>
          <div
            class="do-nothing"
          />
          <div
            class="lbcard-score"
          >
            <h4>
              Asked: 
              2
            </h4>
            <h4>
              Answered: 
              2
            </h4>
            <h4
              class="score"
            >
              Total: 
              4
            </h4>
          </div>
        </div>
        <div
          class="lb-card"
        >
          <div
            class="lbcard-avatar"
          >
            <div
              class="avatar sb-avatar"
              style="display: inline-block; vertical-align: middle; width: 70px; height: 70px; border-radius: 50%; font-family: Helvetica, Arial, sans-serif;"
            >
              <div
                class="avatar sb-avatar__text"
                style="width: 70px; height: 70px; line-height: initial; text-align: center; color: rgb(255, 255, 255); border-radius: 50%;"
                title="Zenobia Oshikanlu"
              >
                <div
                  style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                >
                  <span
                    style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                  >
                    <span />
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="do-nothing"
          />
          <div
            class="lbcard-user"
          >
            <h3>
              Zenobia Oshikanlu
            </h3>
            <h5>
              zoshikanlu
            </h5>
          </div>
          <div
            class="do-nothing"
          />
          <div
            class="lbcard-score"
          >
            <h4>
              Asked: 
              0
            </h4>
            <h4>
              Answered: 
              1
            </h4>
            <h4
              class="score"
            >
              Total: 
              1
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
